src/server/service/kv.rs

```diff
@@ -614,12 +618,17 @@ impl<T: RaftStoreRouter<RocksEngine> + 'static, E: Engine, L: LockManager> Tikv
         stream: RequestStream<RaftMessage>,
         sink: ClientStreamingSink<Done>,
     ) {
+        let store_id = self.store_id;
         let ch = self.ch.clone();
         ctx.spawn(async move {
             let res = stream.map_err(Error::from).try_for_each(move |msg| {
                 RAFT_MESSAGE_RECV_COUNTER.inc();
-                let ret = ch.send_raft_msg(msg).map_err(Error::from);
-                future::ready(ret)
+                if msg.get_to_peer().get_store_id() != store_id {
+                    future::err(Error::from(futures::channel::oneshot::Canceled))
```

Comment:
```suggestion
                    future::err(Error::from(raftstore::Error::StoreNotMatch(to_store_id, my_store_id)))
```
So that L636 could output this mismatch error to log